core/templates/dev/head/collection_editor/CollectionDetailsDirective.js

```diff
@@ -0,0 +1,93 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive displaying and editing a collection details. Edit
+ * options include: chaning the title, changing the objective, changing the
+ * category, and adding a new exploration.
+ *
+ * @author mgowano@google.com (Abraham Mgowano)
+ */
+
+oppia.directive('collectionDetailsDirective', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      getTitle: '&title',
+      getCategory: '&category',
+      getObjective: '&objective',
+      getVersion: '&version',
+      addCollectionNode: '&',
+      setCollectionTitle: '&',
+      setCollectionCategory: '&',
+      setCollectionObjective: '&',
+      addChange: '&'
```

Comment:
Here and in the other editor directives: would it be better for this to come in as a service in the controller, rather than through the $scope? That seems more natural to me -- I'd imagine the change handler service would be a singleton.